1-models/comments-model.js

```diff
@@ -22,23 +22,37 @@ exports.selectCommentsByArticleId = (article_id) => {
 };
 
 exports.insertComment = (username, body, article_id) => {
-   
-    return checkUserExists(username)
-   .then((userExists) => {
+  return checkUserExists(username).then((userExists) => {
     if (!userExists) {
-        return Promise.reject({status:404, message: "User Not Found"})
+      return Promise.reject({ status: 404, message: "User Not Found" });
     }
-    
+
     return selectArticleById(article_id).then(() => {
-        const queryString = `INSERT INTO comments (author, body, article_id) VALUES ($1, $2, $3) RETURNING *`;
-    
-        return db
-          .query(queryString, [username, body, article_id])
-          .then(({ rows }) => {
-            
-            return rows[0];
-          });
-      });
-   })
-    
+      const queryString = `INSERT INTO comments (author, body, article_id) VALUES ($1, $2, $3) RETURNING *`;
+
+      return db
+        .query(queryString, [username, body, article_id])
+        .then(({ rows }) => {
+          return rows[0];
+        });
+    });
+  });
+};
+
+exports.deleteCommentbyId = (comment_id) => {
+  const checkForDeletionQueryString = db.query(
+    `SELECT * FROM comments WHERE comment_id = $1;`,
+    [comment_id]
+  );
+
+  const deleteQueryString = db.query(
+    `DELETE FROM comments WHERE comment_id = $1;`,
+    [comment_id]
+  );
+
+  return Promise.all([checkForDeletionQueryString, deleteQueryString]).then(
+    ([checkResult, deleteResult]) => {
+      return [checkResult, deleteResult];
+    }
+  );
```

Comment:
Your use of a Promise.all here to find if the comment you attempted is one way to solve this problem. Another option would be to use `RETURNING *` at the end of your SQL query. This will return the row you've deleted, so if it's empty your deletion wasn't successful